api/v1/submissions/PKPSubmissionController.php

```diff
@@ -1691,4 +1708,151 @@ protected function getWriteDisabledErrors(string $schemaName, array $params): ar
+        }
+
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_SUBMISSION_FILE, $illuminateRequest->input());
+
+        $submissionFile = Repo::submissionFile()
+            ->addJatsFile(
+                $_FILES['file']['tmp_name'],
+                $_FILES['file']['name'],
+                $submission->getId(),
+                $publication->getId(),
+                SubmissionFile::SUBMISSION_FILE_JATS,
+                $params);
+
+        $jatsFilesProp = $this->getJatsUIProps($request, $submission->getId(), $publication->getId());
+        
+        return response()->json($jatsFilesProp, Response::HTTP_OK);
+    }
+
+    /**
+     * Delete the publication's JATS Submission file
+     */
+    public function deleteJats(Request $illuminateRequest): JsonResponse
+    {
+        $request = $this->getRequest();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+        
+        $publication = Repo::publication()->get((int) $illuminateRequest->route('publicationId'));
+
+        if (!$publication) {
+            return response()->json([
+                'error' => __('api.404.resourceNotFound'),
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return response()->json([
+                'error' => __('api.publications.403.submissionsDidNotMatch'),
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        $submissionFiles = Repo::submissionFile()
+            ->getCollector()
+            ->filterBySubmissionIds([$publication->getData('submissionId')])
+            ->filterByFileStages([SubmissionFile::SUBMISSION_FILE_JATS])
+            ->filterByAssoc(Application::ASSOC_TYPE_PUBLICATION, [$publication->getId()])
+            ->getMany();
+        
+        if ($submissionFiles->isEmpty()) {
+            return response()->json([
+                'error' => __('api.404.resourceNotFound'),
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        $file = $submissionFiles->first();
+
+        Repo::submissionFile()->delete($file);
+
+        $jatsFilesProp = $this->getJatsUIProps($request, $submission->getId(), $publication->getId());
+        
+        return response()->json($jatsFilesProp, Response::HTTP_OK);
+    }
+
+    /**
+     * Helper function to create the props array that returns the appropriate JATS data
+     * either the SubmissionFile Prop or the default JATS content created by the given
+     * submission/publication metadata
+     */
+    protected function getJatsUIProps($request, $submissionId, $publicationId): array
+    {
+        $jatsFile = Repo::submissionFile()
+            ->getJatsFiles($submissionId, $publicationId);
+
+        $jatsFilesProp = [];
+        if (!isset($jatsFile)) {
+            $jatsFilesProp['defaultJatsContent'] = Repo::submissionFile()
+                ->createDefaultContentJatsFile($submissionId, $publicationId);
```

Comment:
This assumes we'll be storing the "default" JATS as a file, right? I think that still needs removal.

Reply:
I used the same function name, which, now, creates the JATS content using the `JatsTemplate` code. I changed the name of the function to `createDefaultJatsContent`